docs/FasterCompileTimes.md

```diff
@@ -0,0 +1,526 @@
+This is Vale's master plan to become the fastest compiling language
+**of all time!**
+
+# Language Changes
+
+## Type-Erasure For Faster Compile Times
+
+### What's Type Erasure?
+
+Java/Scala/etc have type-erasure: as part of compilation, `List<Textends Whatever>` becomes `List<Object>`, because the JVM has no concept of generics.
+
+Typescript also has type-erasure in this way, because Javascript has no
+concept of generics.
+
+They avoid monomorphizing, which speeds up their compilation.
+
+However, nobody's been able to do this with a native language, yet.
+
+Go programmers manually do something **almost** like this. Whenever a Go
+programmer would want a `List<T>`, they instead make a List class which
+uses interface{} (basically, Go's version of a Java Object).
+
+The idea: **Vale should do this automatically in debug mode.** Vale
+would then be the first language with the compilation speed of Go, and
+the run-time speed approaching Rust.
+
+### Why is Type Erasure Faster to Compile?
+
+Type erasure is faster to compile because we only have to generate
+machine code (or JVM, in this case) for `List<Object>`, not all the
+instantiations (`List<Spaceship>`, `List<Boat>`, etc).
+
+If we can hand less functions to LLVM, the entire compiler gets faster.
+See [[Previous Compiler Performance and
+LLVM]{.underline}](https://pling.jondgoodwin.com/post/compiler-performance/),
+which shows how LLVM takes the majority of compilation time. So if we
+can hand less code to LLVM, we speed up compilation time.
+
+### Why can Java and Scala do that, but not C++ and Rust?
+
+#### Because Everything is a Reference
+
+Existing languages (Rust, C++, etc) cannot do this, because generics can
+change the size of an object.
+
+ * In C++, `std::array<Vec3, 2>` is bigger than `std::array<Vec3, 2>`
```

Comment:
I think you made a typo here
`std::array<Vec3, 2>` and `std::array<Vec3, 2>` is the same

Reply:
Fixed, thanks for noticing this!